app/Http/Controllers/PostController.php

```diff
@@ -2,54 +2,86 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Post;
+use App\Models\Group;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Support\Facades\Validator;
 
 class PostController extends Controller
 {
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index()
+    public function __construct()
     {
-        //
+        $this->middleware('auth', ['only' => 'create', 'store', 'destroy']);
     }
 
     /**
-     * Show the form for creating a new resource.
+     * Show the form for creating a new post.
      *
      * @return \Illuminate\Http\Response
      */
-    public function create()
+    public function create($group)
     {
-        //
+        return view('posts.create-post', ['group' => $group]);
     }
 
     /**
-     * Store a newly created resource in storage.
+     * Store a newly created post in storage.
      *
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
     public function store(Request $request)
     {
-        //
+        
+        $request->validate(Post::VALIDATION_RULES);
+        
+        $post = new Post();
+        $user_id = Auth::id();
+        $post->user_id = $user_id;
+        if (! isset($_GET['group'])) {
+            return Response("Not found", 404);
+        }
+        $group_name = $_GET['group'];
+        $group = Group::where('name', $group_name)->first();
+        $post->group_id = $group->id;
+        
+        foreach (Post::VALIDATION_RULES as $field => $_) {
+            $post->$field = $request->$field;
+        }
+
+        if (isset($request['image'])) {
+            $post['image'] = $request['image'];
+        }
+
+        $post->save();
+
+        return redirect()->route('post.show', ['id' => $post->id]);
     }
 
     /**
-     * Display the specified resource.
+     * Display the specified post.
      *
      * @param  int  $id
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\View\View|\Illuminate\Http\Response
      */
     public function show($id)
     {
-        //
+        $validator = Validator::make(['id' => $id], [
+            'id' => 'required|int|exists:posts'
+        ]);
+
+        if ($validator->fails()) {
+            return Response("Not found", 404);
+        }
```

Comment:
Right know we do it this way in the other controllers.
We can decide which method we want to keep but let's use the same one if we can.

```suggestion
        Validator::validate(['id' => $id], [
            'id' => 'required|int|exists:posts'
        ]);
```